### PR TITLE
Initial checkin of Metrics LineGraph Component

### PR DESCRIPTION
**Pull request by mrtracy**
Commit creates the first version of a reusable LineGraph component, which graphs
the results of cockroach time series query.

It's a bit bare-bones in appearance, but is already reusable and should provide
a good foundation for building our admin monitor pages.

### CODE REVIEW

**Comment by BramGruneir** [comment]
Can you add a screenshot?

File: resource/ui/index.html [hunk 1]
```diff
@@ -23,6 +23,7 @@
     <link href='https://fonts.googleapis.com/css?family=Source+Sans+Pro:400,900|Source+Code+Pro' rel='stylesheet' type='text/css'>
     <link rel="stylesheet" href="/css/app.css">
     <link rel="stylesheet" href="/css/rest_explorer.css"> <!-- TODO(andybons): @import-like behavior -->
+    <link rel="stylesheet" href="/css/graph.css"> <!-- TODO(andybons): @import-like behavior -->
```
Comment (BramGruneir) [inline resource/ui/index.html]: drop the todo

File: resource/ui/ts/components/metrics.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+                // Line data interpreter.
+                private line = d3.svg.line()
+                    .x((d) => this.timeScale(d.timestamp_nanos/1.0e6))
+                    .y((d) => this.valScale(d.value));
+
+                private color = d3.scale.category10();
+
+                // Query results.
+                results;
```
Comment (BramGruneir) [inline resource/ui/ts/components/metrics.ts]: add types

File: resource/ui/ts/components/metrics.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                error = m.prop("");
```
Comment (BramGruneir) [inline resource/ui/ts/components/metrics.ts]: add type

File: resource/ui/ts/components/metrics.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+            /**
+             * create instantiates a single instance of the LineGraph component
+             * which displays data from the supplied Metrics.Query object.
+             *
+             * @param key The key param is used by mithril to track objects in lists which can be rearranged.
+             */
+            export function create(width:number, height:number, query:Models.Metrics.Query, key?:number){
+                var vm:ViewModel = {width:width, height:height, query:query}
+                if (!!key) {
```
Comment (BramGruneir) [inline resource/ui/ts/components/metrics.ts]: ugg...  do you prefer some of these weird js operations?

**Comment by BramGruneir** [comment]
LGTM

File: resource/ui/ts/components/metrics.ts (hunk 4 above)
Comment (mrtracy) [inline resource/ui/ts/components/metrics.ts]: I like !!

**Comment by spencerkimball** [comment]
Matt, I think we shouldn't spend time doing our own graphs!

Let's use: http://c3js.org/

**Comment by spencerkimball** [comment]
There's also http://nvd3.org/

and http://fastly.github.io/epoch/
The timeseries-based real time charts are cool: http://fastly.github.io/epoch/real-time/

This one is pretty amazing: http://dc-js.github.io/dc.js/

Not bad for time series info: http://square.github.io/cubism/

**Comment by mrtracy** [comment]
Using a library is the plan, this checkin was mostly to #1. Get the data from our endpoint and #2. Make a mithril component that gets the data in a predictable way.

The d3 stuff was just copy/pasted from one of d3's line chart demos.  I spent some time to understand how the library worked at a basic level, but I was planning to go with the [rickshaw](http://code.shutterstock.com/rickshaw/) library.

**Pull request merged**